main/modes/games/swadgedoku/swadgedoku.c

```diff
@@ -0,0 +1,1370 @@
+                                               .type        = TROPHY_TYPE_PROGRESS,
+                                               .difficulty  = TROPHY_DIFF_EXTREME,
+                                               .maxVal      = (int)SD_HARDEST + 1,
+                                               .identifier  = &anyPuzzleTrigger,
+                                           },
+                                           {
+                                               .title       = "Revised rules",
+                                               .description = "Play each Sudoku variant",
+                                               .image       = NO_IMAGE_SET,
+                                               .type        = TROPHY_TYPE_CHECKLIST,
+                                               .difficulty  = TROPHY_DIFF_HARD,
+                                               .maxVal      = (1 << ((int)SM_X_GRID + 1)) - 1,
+                                               .identifier  = &anyPuzzleTrigger,
+                                           }};
+
+// Individual mode settings
+trophySettings_t swadgedokuTrophySettings = {
+    .drawFromBottom   = false,
+    .staticDurationUs = DRAW_STATIC_US * 2,
+    .slideDurationUs  = DRAW_SLIDE_US,
+};
+
+// This is passed to the swadgeMode_t
+trophyDataList_t swadgedokuTrophyData
+    = {.settings = &swadgedokuTrophySettings, .list = swadgedokuTrophies, .length = ARRAY_SIZE(swadgedokuTrophies)};
+
+// Aliases for trophies
+const trophyData_t* trophySolveAny = &swadgedokuTrophies[0];
+const trophyData_t* trophyTenMins  = &swadgedokuTrophies[1];
+const trophyData_t* trophyFiveMins = &swadgedokuTrophies[2];
+
+//==============================================================================
+// Variables
+//==============================================================================
+
+swadgeMode_t swadgedokuMode = {
+    .modeName                 = swadgedokuModeName,
+    .wifiMode                 = NO_WIFI,
+    .overrideUsb              = false,
+    .usesAccelerometer        = false,
+    .usesThermometer          = false,
+    .overrideSelectBtn        = false,
+    .fnEnterMode              = swadgedokuEnterMode,
+    .fnExitMode               = swadgedokuExitMode,
+    .fnMainLoop               = swadgedokuMainLoop,
+    .fnAudioCallback          = NULL,
+    .fnBackgroundDrawCallback = swadgedokuBackgroundDrawCallback,
+    .fnEspNowRecvCb           = NULL,
+    .fnEspNowSendCb           = NULL,
+    .fnAdvancedUSB            = NULL,
+    .fnDacCb                  = NULL,
+    .trophyData               = &swadgedokuTrophyData,
+};
+
+static swadgedoku_t* sd = NULL;
+
+//==============================================================================
+// Functions
+//==============================================================================
+
+static void swadgedokuEnterMode(void)
+{
+    sd = calloc(1, sizeof(swadgedoku_t));
+    if (!sd)
+    {
+        // make cppcheck happy
```

Comment:
Huh, I've never noticed it complain about this. I guess I only have v2.14.0. Smart though.